tests/unit/test_xtce/test_definitions.py

```diff
@@ -541,3 +541,44 @@ def test_parse_methods(test_data_dir):
     empty_packet = spp.SpacePacket(binary_data=empty_packet_data)
     with pytest.warns(DeprecationWarning, match="parse_ccsds_packet is deprecated"):
         assert packet == xdef.parse_ccsds_packet(empty_packet)
+
+
+def test_parse_packet_extra_bytes(test_data_dir):
+    """Test parsing a packet that has too many raw bytes
+
+    This should warn the user that there is unparsed data
+    """
+    xdef = definitions.XtcePacketDefinition.from_xtce(test_data_dir / "test_xtce.xml")
+
+    # Test parsing a packet that is longer than the definition
+    too_long_packet_data = space_packet_parser.ccsds.create_ccsds_packet(data=bytes(70),
+                                                    apid=11,
+                                                    sequence_flags=space_packet_parser.ccsds.SequenceFlags.UNSEGMENTED)
+
+    # Full packet object with read methods attached
+    empty_packet = spp.SpacePacket(binary_data=too_long_packet_data)
+
+    with pytest.warns(UserWarning,
+                      match=r"Number of bits parsed \(568b\) did not match the length of data available \(608b\)"):
+        xdef.parse_packet(empty_packet)
```

Comment:
This was added previously. Maybe we should move that out into a separate test rather than buried though. https://github.com/lasp/space_packet_parser/pull/174/files#diff-d00a6420b1a4109fde4e4cabbaa51a66b0056a9f9bc21ea042495fb3f3d3f6b9R530

What would you think about deprecating `parse_packet` and making that a private method if at all? The parse_bytes() should really be the entrypoint and then we construct an empty packet for the user to then iterate through the content.